asp_interactive_configuration/examples/metro/fconstraints.lp

```diff
@@ -0,0 +1,79 @@
+% Copyright (c) 2022 Siemens AG Oesterreich
+% SPDX-License-Identifier: MIT
+
+% TODO: still missing is the length_mm attribute
+%
+#program domain(new_object, cls).
+
+ooasp_potential_cv(seats_eq__nrseats).
+ooasp_potential_cv(stand_iff_handrail).
+
+% Paggensers number is calculated
+ooasp_cv(nrpassengers_sum,new_object,"Number of passenges not adding up",(new_object,)):-
+    ooasp_isa(wagon,new_object),
+    &fsum{ooasp_attr_fvalue(nr_seats,new_object);
+          ooasp_attr_fvalue(standing_room,new_object)}!=
+        ooasp_attr_fvalue(nr_passengers,new_object).
+
+%  Only one wagon allowed
+ooasp_cv(only_one_wagon,new_object,"Only one wagon allowed, found also wagon {}",(W2,)):-
+    ooasp_isa(wagon,new_object),
+    ooasp_isa(wagon,W2),
+    new_object>W2.
+
+% Handrail required if standing_room>0
+ooasp_cv(stand_iff_handrail,W,"Handrail required if standing room is larger than 0",(new_object,)):-
+    active(new_object),
+    ooasp_isa(wagon,W),
+    #count{ID2:ooasp_assoc_gen(wagon_handrail,OPT,W,ID2)}=0,
+    &fsum{ooasp_attr_fvalue(standing_room,W)}>0.
+
+
+% the number of associated seats must be equal to the attribute nr_seats
+ooasp_cv(seats_eq__nrseats,W,"Number of seats should be {}, since there are {} Seats associated",(S,new_object)):-
```

Comment:
The explanation given here is wrong. The second {} will be bound to new_object, which for nr_seats = 10, new_object=1 and no associated seats results in a message like:
"Number of seats should be 0, since there are 1 Seats associated"